src/Monocle/Api/Server.hs

```diff
@@ -716,6 +729,31 @@ instance TrendPB Word32 where
       . MetricPB.HistoIntStat
       $ Q.toPBHistoInt <$> v
 
+class Num a => TopPB a where
+  toTopResult :: Maybe (Q.TermsCount a) -> MetricPB.GetResponse
+
+toTopResultNotSupported :: MetricPB.GetResponse
+toTopResultNotSupported =
+  MetricPB.GetResponse
+    . Just
+    $ MetricPB.GetResponseResultError "This metric does not support Top option"
```

Comment:
That's a bit odd, shouldn't it be:
```suggestion
  toTopResult :: Q.TermsCount a -> MetricPB.GetResponse
  
toTopResultOrFail :: Maybe (Q.TermsCount a) -> MetricPB.GetResponse
toTopResultOrFail = \case
  Just tc -> toTopResult tc
  Nothing -> MetricPB.GetResponse
    . Just
    $ MetricPB.GetResponseResultError "This metric does not support Top option"
```

And then in metricGet, use `toTopResultOrFail`. That way we don't have to handle the missing result in each instance.